rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -149,31 +149,46 @@ def generate_zero_string(membset, fill_args):
 @[if not member_list]@
     (void)_init;
 @[end if]@
-@[for membset in member_list]@
-@[ if membset.members[0].default_value is not None]@
+@{
+members_if_all_defaults = [m for m in member_list if m.members[0].default_value]
+members_else_if_zero = [m for m in member_list if m.members[0].zero_value]
+members_if_all_zero = [
+    m for m in member_list if  (m.members[0].zero_value or m.members[0].zero_need_array_override)
+    and not m.members[0].default_value
+]
+}@
+@[if members_if_all_defaults]@
     if (rosidl_generator_cpp::MessageInitialization::ALL == _init ||
       rosidl_generator_cpp::MessageInitialization::DEFAULTS_ONLY == _init)
     {
+@[for membset in members_if_all_defaults]@
```

Comment:
The indentation is off. Each em block should be indented one level - 2 spaces.

Same throughout the patch.